ee/debug/checkups/intune_windows.go

```diff
@@ -0,0 +1,95 @@
+//go:build windows
+// +build windows
+
+package checkups
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type intuneCheckup struct {
+	summary string
+}
+
+func (i *intuneCheckup) Name() string {
+	return "Intune"
+}
+
+func (i *intuneCheckup) Run(ctx context.Context, extraWriter io.Writer) error {
+	// Other areas of interest: https://learn.microsoft.com/en-us/mem/intune/remote-actions/collect-diagnostics
+
+	zipWriter := zip.NewWriter(extraWriter)
+	defer zipWriter.Close()
+
+	if err := agentLogs(zipWriter); err != nil {
+		i.summary += fmt.Sprintf("Failed to collect Intune agent logs: %v. ", err)
+	}
+
+	if err := installLogs(zipWriter); err != nil {
+		i.summary += fmt.Sprintf("Failed to collect Intune install logs: %v. ", err)
+	}
+
+	i.summary = strings.TrimSpace(i.summary)
+
+	return nil
+}
+
+func agentLogs(zipWriter *zip.Writer) error {
+	agentLogsPathPattern := filepath.Join(os.Getenv("SYSTEMROOT"), "ProgramData", "Microsoft", "IntuneManagementExtension", "Logs", "*")
+	matches, err := filepath.Glob(agentLogsPathPattern)
+	if err != nil {
+		return fmt.Errorf("globbing for agent logs at %s: %w", agentLogsPathPattern, err)
+	}
+	if len(matches) == 0 {
+		return fmt.Errorf("no intune agent logs found at %s", agentLogsPathPattern)
+	}
```

Comment:
Is this an error? (It will run on non-intune machines)

Reply:
I think I just added this to make it more obvious what was going on when testing -- I'll update to remove here + below

Reply:
I think I was mostly quibbling over `Errorf` vs some debug/info log line, I don't really want to see the error on every flare...